emukit/core/acquisition/integrated_acquisition.py

```diff
@@ -3,7 +3,7 @@
 import numpy as np
 
 from emukit.core.acquisition import Acquisition
-from emukit.core.interfaces import IModel, IPriorHyperparameters, IDifferentiable
```

Comment:
Isn't it actually used? I mean, there is definitely a call of `model.evaluate_with_gradients` below. I'd suggest to add this interface to model type hint

Reply:
I have always been confused how we should do type hints when some interfaces are optional to implement. The type hint should say "Must implement IModel and optionally implement IDifferentiable". We usually do `Union[IModel, IDifferentialbe]` which means "must implement both IModel and IDifferentialbe".  You get warnings from the type checking system if you pass in a model that doesn't implement gradients with the current type hints.